File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a,b):
    while b>0:
        a,b = b,a%b
    return a

def lcm(a,b):
    return a*b // lcm(a,b)

def solution(n, m):
    answer = []
    temp1,temp2 = n,m
    while m > 0:
        n,m = m,n%m
    answer.append(n)
    answer.append(temp1*temp2 // n)
    return answer

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    return [gcd(n,m),lcm(n,m)];
}

function gcd(a,b){
    if(a%b ==0){
        return b;
    }
    return gcd(b,a%b);
}

function lcm(a,b){
    return (a*b) / gcd(a,b);
}